packages/@aws-cdk/aws-ecs-patterns/lib/base/queue-processing-service-base.ts

```diff
@@ -191,8 +204,23 @@ export abstract class QueueProcessingServiceBase extends Construct {
     }
     this.cluster = props.cluster || this.getDefaultCluster(this, props.vpc);
 
-    // Create the SQS queue if one is not provided
-    this.sqsQueue = props.queue !== undefined ? props.queue : new Queue(this, 'EcsProcessingQueue', {});
+    // Create the SQS queue if one is not provided, when creating SQS queue setup DLQ
+    if (props.queue) {
+      this.sqsQueue = props.queue;
+    } else {
+      this.deadLetterQueue = new Queue(this, "EcsProcessingDeadLetterQueue", {
+        retentionPeriod: Duration.days(14)
+      });
+      this.sqsQueue = new Queue(this, 'EcsProcessingQueue', {
+        deadLetterQueue: {
+          queue: this.deadLetterQueue,
+          maxReceiveCount: props.maxReceiveCount || 5
```

Comment:
Curious as to why you decided to default to 5 (seems like a magic number). If this is not a mandatory property, I would lean towards not defining a default, but rather only propagating the property if it's passed in.

Reply:
i don't think maxReceiveCount is an optional attribute https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-sqs/lib/queue.ts#L142-L155